pages/search/index.js

```diff
@@ -1,48 +1,308 @@
 import styled from "styled-components";
 import VerticalGameList from "@/components/VerticalGameList";
 
+import Fuse from "fuse.js";
+
 import { useState } from "react";
 import useSWR from "swr";
+import { FaTimes } from "react-icons/fa";
+
+const INITIAL_DATA = {
+  name: "",
+  categories: "",
+  rating: "",
+  minAge: "",
+  players: "",
+  yearpublished: "",
+  playtime: "",
+};
 
-export default function SearchPage({ isFavorite, toggleFavorite, showModal }) {
+export default function SearchPage({ isFavorite, toggleFavorite }) {
   const { data } = useSWR("./api/games");
-  const [searchInput, setSearchInput] = useState("");
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const [showFilters, setShowFilters] = useState(false);
+  const [filteredResults, setFilteredResults] = useState([]);
```

Comment:
you don't need this state. Derived states can be computed as normal constants since if the filters change the entire component is executed again, and therefore any code that comes below the state statements. In this case it would be the chained .filter methods. this can look like this: 
```js
const [filters, setFilters] = useState(INITIAL_FILTERS);
const results = filterGames(data);

function filterGames(games) {
  const filteredForName = games.filter(game => game.startsWith(filters.name))
  const filteredForCategories = filteredForName.filter(game => ...)
  return // final results
}

```